src/mpid/ch4/netmod/ofi/ofi_rma.h

```diff
@@ -462,7 +462,7 @@ static inline int MPIDI_NM_mpi_rput(const void *origin_addr,
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_NM_MPI_RPUT);
     int mpi_errno = MPI_SUCCESS;
 
-    if (!MPIDI_OFI_ENABLE_RMA) {
+    if (!MPIDI_OFI_ENABLE_RMA || !MPIDI_OFI_WIN(win).mr) {
         mpi_errno = MPIDIG_mpi_rput(origin_addr, origin_count, origin_datatype, target_rank,
```

Comment:
Emm, after thinking this again, the per RMA check seems incorrect too. `MPIDI_OFI_WIN(win).mr == NULL` means the local base is NULL. But what we really want to check is whether the remote MR is NULL. Please check `MPIDI_OFI_winfo_mr_key`.

Perhaps it is better to just check `win->create_flavor`. For other non-dynamic windows, the remote MR is always created as long as the remote winsize is non-zero.

Reply:
I see. I'll follow up.

Reply:
fixed.